OpenSim/Common/TableSource.h

```diff
@@ -0,0 +1,214 @@
+        msg += " timestamp = " + std::to_string(timestamp);
+
+        addMessage(msg);
+    }
+}; // class TimeOutOfRange
+
+
+/** Component representing a source of data from a TimeSeriesTable_.
+
+This Component has two outputs:
+- A list output wih one channel per column of the TimeSeriesTable_. 
+- A non-list output for a row of the TimeSeriesTable_.
+
+Construct this Component by giving it a TimeSeriesTable_. Then use it by 
+connecting its output to the Input of another Component that accepts compatible
+type of Input. Make sure to populate the column-labels of the TimeSeriesTable_ 
+before connecting this Component to the input of another Component.    
+
+\tparam ET Type of each element of the TimeSeriesTable_ this Component 
+           represents.                                                        */
+template<typename ET>
+class TableSource_ : public Component {
+    OpenSim_DECLARE_CONCRETE_OBJECT_T(TableSource_, ET, Component);
+
+public:
+    /** Type of the TimeSeriesTable_ this Component will hold.                */
+    typedef TimeSeriesTable_<ET> Table;
+    /** Type of the 'row' Output of this Component.                           */
+    typedef SimTK::Vector_<ET>   Vector;
+
+    OpenSim_DECLARE_OUTPUT(row, Vector, getRowAtTime, 
+                           SimTK::Stage::Time);
+    OpenSim_DECLARE_LIST_OUTPUT(column, ET, getColumnAtTime, 
+                                SimTK::Stage::Time);
+
+    TableSource_()                               = default;
+    TableSource_(const TableSource_&)            = default;
+    TableSource_(TableSource_&&)                 = default;
+    TableSource_& operator=(const TableSource_&) = default;
+    TableSource_& operator=(TableSource_&&)      = default;
+    
+    /** Construct the TableSource_ by giving it a TimeSeriesTable_ to hold.   */
+    TableSource_(const Table& table) : 
+        _table{table} 
+    {}
+
+    /// \name Get/Set underlying TimeSeriesTable_
+    /// @{
+
+    /** Get a read-only reference to the TimeSeriesTable_ this TableSource_ 
+    currently holds.                                                          */
+    const Table& getTable() const {
+        return _table;
+    }
+
+    /** Replace the existing TimeSeriesTable_ this TableSource_ currently 
+    holds.                                                                    
+
+    \throws KeyNotFound If table provided does not have column-labels.        */
+    void setTable(const Table& table) {
+        _table = table;
+        auto& columnOutput = updOutput("column");
+        columnOutput.clearChannels();
+        for(const auto& columnLabel : _table.getColumnLabels())
+            columnOutput.addChannel(columnLabel);
+    }
+
+    /// @}
+
+protected:
+    /** Retrieve value of a column at a given time(implicit in the State 
+    provided). Linear interpolation is performed if the TimeSeriesTable_ does
+    not contain an entry for the time mentioned by the state.
```

Comment:
I'm confused. It seems from this comment that `getColumnAtTime()` returns the value from exactly one column (whose label is specified as the second argument) at exactly one time. But then

``` cpp
OpenSim_DECLARE_LIST_OUTPUT(column, ET, getColumnAtTime, SimTK::Stage::Time);
```

calls this method and assembles a list somehow. It would be helpful if the doxygen for this method provided some explanation. :icecream: